### ISSUE TITLE
Greedy/possessive optimizer fails on negative lookahead

### ISSUE THREAD

**Issue by Alcaro** [hanickadot/compile-time-regular-expressions#74]
```
#include <ctre.hpp>

static constexpr auto re = ctll::fixed_string{"a+(?!b)"};
bool x(const char * a, const char * b)
{
	return ctre::match<re>(a, b);
}
```

```
ctre/evaluation.hpp: In instantiation of 'constexpr R ctre::evaluate(Iterator, Iterator, EndIterator, R, ctll::list<ctre::repeat<A, B, Content ...>, Tail ...>) [with R = ctre::regex_results<const char*>; Iterator = const char*; EndIterator = const char*; long unsigned int A = 1; long unsigned int B = 0; Content = {ctre::character<'a'>}; Tail = {ctre::sequence<ctre::lookahead_negative<ctre::character<'b'> > >, ctre::assert_end, ctre::end_mark, ctre::accept}]':
ctre/evaluation.hpp:297:24: error: no matching function for call to 'collides(ctll::list<ctre::character<'a'> >, ctre::can_be_anything)'
  297 |  if constexpr (collides(calculate_first(Content{}...), calculate_first(Tail{}...))) {
ctre/first.hpp:470:56: note: candidate: 'template<class ... A, class ... B> constexpr bool ctre::collides(ctll::list<Ts ...>, ctll::list<Bs ...>)'
  470 | template <typename... A, typename... B> constexpr bool collides(ctll::list<A...> rhs, ctll::list<B...> lhs) {
ctre/first.hpp:470:56: note:   template argument deduction/substitution failed:
ctre/evaluation.hpp:297:24: note:   'ctre::can_be_anything' is not derived from 'ctll::list<Bs ...>'
```

https://godbolt.org/z/THZW_t

Setting CTRE_DISABLE_GREEDY_OPT makes it compile as expected.

**Comment by hanickadot** [comment]
Fixed in 274202fc2f88c9f76a18685ed80e36f92780a3f1.

**Comment by hanickadot** [comment]
Thank you for reporting the issue, if you find another, please don't hesitate to report it :)
